### PR TITLE
KOGITO-523: [DMN Designer] Cannot define Time Data Type Range constraint

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/BAPL-1376

(Although manifested itself in https://issues.jboss.org/browse/KOGITO-523).

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/858
- https://github.com/kiegroup/kie-wb-common/pull/2990

### CODE REVIEW

**Comment by danielzhe** [comment]
@manstis My only concern with this approach is localization to Japanese. I don't know if they use "our alphabet" to timezones or their alphabet, otherwise I think that is ok because is universal and I don't think that create a new timezone is something common.

**Comment by manstis** [comment]
@danielzhe Could you please explain your concern a little more?

Are you saying the server side approach returns i18n'ed texts? I didn't see anywhere the User's locale we considered when extracting the time zone information!? Perhaps I misunderstand.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@danielzhe @karreiro @jomarko This PR is about as good as it's going to get.. ready for review!

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Much of the code is generated..... I'm not going to make the generation accommodate repeated literals.

**Comment by manstis** [comment]
Hi @jomarko let me try to explain this PR a bit better.

The Business Central editor has a _real_ Java backend and this uses `TimeZoneServiceImpl` to retrieve a list of time zones from the Java Runtime Environment (from a `tzdb.dat` in your PC's `jre/lib` folder). This is not available to kogito that runs 100% in the browser.

So, class `TimeZoneDumper` can be ran from the CLI (or in an IDE etc) and this simply outputs the results of `TimeZoneServiceImpl` to _stdio_ (`System.out`) as a `String`. I then simply (and manually) copy this `String` and paste it into the new class used by kogito namely [TimeZones](https://github.com/kiegroup/kie-wb-common/blob/02f10ce17a307821e053ebae91078026b6e80724/kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/TimeZones.java).

Time zones are pretty static and I think this approach is _good enough_. An alternative approach would be to generate the content of the `TimeZones` class at GWT compilation time but to be honest I felt it was over engineered for what is in essence a static list of time zones.

Does this help explain things?

**Comment by jomarko** [comment]
Ok, I understand now probably that `TimeZoneDumper` is helper class that prints content of the `TimeZone.getTimeZones` however I do not see added value in tracking that file.

It requires additional configuration for maven plugins to not have it in jars. Even if Java.FUTURE will provide more timezones, our `Dumper` is not started automatically, its output is compared in any test. 

Maybe I am missing something, but I do not see the reason why to track this file.

**Comment by manstis** [comment]
@jomarko I understand your comment and will happily remove.

However before doing so would like to hear @karreiro and @danielzhe opinions too.

**Comment by manstis** [comment]
@karreiro I'll take a look at Moment Timezone :-)

**Comment by manstis** [comment]
@jomarko @danielzhe This now uses https://momentjs.com/timezone to get client-side time-zones.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/DMNKogitoCommonEntryPoint.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.jboss.errai.ioc.client.api.EntryPoint;
+import org.uberfire.client.views.pfly.sys.PatternFlyBootstrapper;
+
+@EntryPoint
+public class DMNKogitoCommonEntryPoint {
+
+    @PostConstruct
+    public void init() {
+        PatternFlyBootstrapper.ensureMomentTimeZoneIsAvailable();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/DMNKogitoCommonEntryPoint.java]: @manstis sorry for my question, could you please explain/remind me, why we load moment time zone in this class, and moment just in `kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/DMNClientEntryPoint` I mean in totally different module.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/TimeZonesProvider.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        return TIME_ZONES;
+    }
+
+    private List<DMNSimpleTimeZone> buildTimeZones() {
+        final List<DMNSimpleTimeZone> timeZones = new ArrayList<>();
+        final String[] names = getNames();
+
+        for (String name : names) {
+            final double offset = getOffset(name);
+            final String format = getOffsetString(name);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/TimeZonesProvider.java]: Personal opinion, it is confusing to mix `format` and `offsetString`, I mean third parameter of `DMNSimpleTimeZone` 

feel free to ignore if you think it is fine

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/DMNKogitoCommonEntryPoint.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/DMNKogitoCommonEntryPoint.java]: Hi @jomarko moment time zone is only used by the _kogito_ editors (to retrieve a list of time-zones in the _time_ related data-type constraint editor popups). moment is used by both Business Central and _kogito_ to handle _date_.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/DMNKogitoCommonEntryPoint.java]: This _kogito_ related module depends on `kie-wb-common-dmn-client` and hence for _kogito_ editors both moment and moment time zone are loaded/injected into the HTML page.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/TimeZonesProvider.java (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/TimeZonesProvider.java]: Updated... I orginally had a method called `getFormat(..)` that is consistent with moment time zone but then I saw `DMNSimpleTimeZone` called it `offsetString` so I changed the method name for consistency; forgetting to change the variable name.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**